internal/app/server/handlers/handler.go

```diff
@@ -156,6 +181,72 @@ func (h *Handler) createFromBatch(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *Handler) getURLsByUser(w http.ResponseWriter, r *http.Request) {
+	_, err := r.Cookie(middlewareUtils.CookieName)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	userID := contextUtils.GetUserID(r.Context())
+	urls, err := h.urlShortener.GetByUserID(r.Context(), userID)
+	if err != nil {
+		logger.Log.Error("error to get url")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	var shortURLBatch []ShortenUserResponseBatch
+	for _, url := range *urls {
+		shortURLBatch = append(shortURLBatch, ShortenUserResponseBatch{
+			ShortURL:    h.prefixURL + url.ShortURL,
+			OriginalURL: url.OriginalURL,
+		})
+	}
+
+	if len(shortURLBatch) == 0 {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	err = json.NewEncoder(w).Encode(shortURLBatch)
+	if err != nil {
+		logger.Log.Error("error to create response", zap.String("err", err.Error()))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+}
+
+func (h *Handler) deleteShorURLs(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	var shortURLs []string
+	err := json.NewDecoder(r.Body).Decode(&shortURLs)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	userID := contextUtils.GetUserID(r.Context())
+	h.deleteJobs <- deleteJob{
+		shortURL: shortURLs,
+		userID:   userID,
+	}
```

Comment:
Это скорее fan-out, чем fan-in. Суммарно, исходя из входящих условий данное решение способно асинхронно обработать 5 одновременных запросов на удаление. Если придет больше, то один из запросов будет вынужден  дождаться, пока освободится один из воркеров. Для реализации реального fain-in необходимо реализовать нечто такое:
```suggestion
	go func() {
		h.deleteJobs <- deleteJob{
			shortURL: shortURLs,
			userID:   userID,
		}
	}()
```
В остальном все сделано вполне верно.